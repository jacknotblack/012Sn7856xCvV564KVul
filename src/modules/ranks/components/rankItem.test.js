import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RankItem from "./rankItem";

describe("A suite", () => {
  let wrapper;
  const props = {
    rankNumber: 4,
    name: "Tony Stark",
    stars: 5167456314675
  };

  beforeEach(() => {
    wrapper = shallow(<RankItem {...props} />);
  });
  it("renders without crashing", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render name", () => {
    expect(wrapper.find(".name").text()).toEqual(props.name);
  });

  it("should render star numbers", () => {
    expect(wrapper.find(".star-wrapper").text()).toEqual(
      props.stars.toString()
    );
  });

  it("should render rank numbers", () => {
    expect(wrapper.find(".rank-number").text()).toEqual(
      `No. ${props.rankNumber}`
    );
  });

  it("should add class according to rank number to rank-number div", () => {
    expect(wrapper.find(".rank-number").hasClass(`no${props.rankNumber}`));
  });
});
