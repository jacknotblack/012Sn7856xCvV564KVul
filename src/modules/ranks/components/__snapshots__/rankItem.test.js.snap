// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite should match snapshot 1`] = `
<div
  className="rank-item"
>
  <div
    className="rank-number no4"
  >
    No. 
    4
  </div>
  <div
    className="avatar-container"
  >
    <img
      alt="avatar"
      className="avatar"
      src="avatar.jpg"
    />
    <div
      className="name-wrapper"
    >
      <span
        className="name"
      >
        Tony Stark
      </span>
      <div
        className="star-wrapper"
      >
        5167456314675
      </div>
    </div>
  </div>
</div>
`;
